plugins/inputs/suricata/suricata.go

```diff
@@ -0,0 +1,229 @@
+package suricata
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	// InBufSize is the input buffer size for JSON received via socket.
+	// Set to 10MB, as depending on the number of threads the output might be
+	// large.
+	InBufSize = 10 * 1024 * 1024
+)
+
+// Suricata is a Telegraf input plugin for Suricata runtime statistics.
+type Suricata struct {
+	sync.Mutex
+
+	Source    string `toml:"source"`
+	Delimiter string `toml:"delimiter"`
+
+	InputListener *net.UnixListener
+	Ctx           context.Context
+	Cancel        context.CancelFunc
```

Comment:
Change the fields that shouldn't be settable from TOML to lowercase, otherwise they could be set with `ctx = "foo"`.

Reply:
Done.